```
				{
					"name": "test",
					"state": {
						"terminated": {
							"exitCode": 2,
							"reason": "Error",
							"message": " no such host\"\nUnable to connect to the server: dial tcp: lookup api.e2e-rotate.local.external.local.gardener.cloud on 127.0.0.1:53: no such host\n\u003e Exporting events of shoot cluster 'shoot--garden--e2e-managedseed'\nmake[1]: Entering directory '/home/prow/go/src/github.com/gardener/gardener'\n/home/prow/go/src/github.com/gardener/gardener/dev-setup/seed.sh down\nDetected scenario: single-node\nUsing skaffold profile: single-node\nCleaning up...\n - secret \"local.login\" deleted from garden namespace\n - secret \"seed-local\" deleted from garden namespace\n - gardenlet.seedmanagement.gardener.cloud \"local\" deleted from garden namespace\nError from server (Forbidden): seeds.core.gardener.cloud \"local\" is forbidden: cannot delete seed local since it is still used by shoot(s)\nmake[1]: *** [Makefile:346: seed-down] Error 1\nmake[1]: Leaving directory '/home/prow/go/src/github.com/gardener/gardener'\nmake: *** [Makefile:406: ci-e2e-kind] Error 2\nwrapper.sh] [TEST] Test Command exit code: 2\nwrapper.sh] [CLEANUP] Waiting 30 seconds for pods stopped with terminationGracePeriod:30\nwrapper.sh] [CLEANUP] Cleaning up after Docker in Docker ...\n1c01bb672634\n14e6591df31f\n9bcca58bfefc\n517e301c5a66\n40608d4a7aec\nae3f257cc1fe\ndd212764fc9a\n0582f8e4956f\n17bebf770222\ne12c5abb6b25\nbbdccd65f078\n0b757560fe8c\na3436314340a\n35b650c69385\n99fd1123076b\nd9e97dab66fb\nbb4c5f16ef88\n332b7dff5b2e\n360e40b39735\n7fc09b7954f0\nba8b69ff9a25\n7dfb1e41887a\nwrapper.sh] [CLEANUP] Waiting for docker to stop for 30 seconds\nStopping Docker: dockerProgram process in pidfile '/var/run/docker-ssd.pid', 1 process(es), refused to die.\nwrapper.sh] [CLEANUP] Done cleaning up after Docker in Docker.\n================================================================================\nwrapper.sh] Exiting 2\n{\"component\":\"entrypoint\",\"error\":\"wrapped process failed: exit status 2\",\"file\":\"sigs.k8s.io/prow/pkg/entrypoint/run.go:84\",\"func\":\"sigs.k8s.io/prow/pkg/entrypoint.Options.internalRun\",\"level\":\"error\",\"msg\":\"Error executing test process\",\"severity\":\"error\",\"time\":\"2026-06-17T16:24:05Z\"}\n",

```

```
STEP: Verify that gardenlet's kubeconfig secret has actually been renewed @ 06/17/26 16:19:38.765
  [TIMEDOUT] in [It] - /home/prow/go/src/github.com/gardener/gardener/test/e2e/gardener/managedseed/create_rotate_delete.go:238 @ 06/17/26 16:21:38.765
  A running node failed to exit in time
  Ginkgo is moving on but a node has timed out and failed to exit before its grace period elapsed.  The node has now leaked and is running in the background.
  Here's a current progress report:
    ManagedSeed Tests Trigger gardenlet kubeconfig rotation by annotating Seed Should verify after (Spec Runtime: 2m30.003s)
      /home/prow/go/src/github.com/gardener/gardener/test/e2e/gardener/managedseed/create_rotate_delete.go:238
      In [It] (Node Runtime: 2m30.003s)
        /home/prow/go/src/github.com/gardener/gardener/test/e2e/gardener/managedseed/create_rotate_delete.go:238
        At [By Step] Verify that gardenlet's kubeconfig secret has actually been renewed (Step Runtime: 2m30.003s)
          /home/prow/go/src/github.com/gardener/gardener/test/utils/rotation/gardenlet_kubeconfig.go:75
        Spec Goroutine
        goroutine 104 [select]
          github.com/onsi/gomega/internal.(*AsyncAssertion).match(0x28a8a54bca80, {0x35b6ab0, 0x57e5540}, 0x1, {0x0, 0x0, 0x0})
            /home/prow/go/pkg/mod/github.com/onsi/gomega@v1.42.0/internal/async_assertion.go:558
          github.com/onsi/gomega/internal.(*AsyncAssertion).Should(0x28a8a54bca80, {0x35b6ab0, 0x57e5540}, {0x0, 0x0, 0x0})
            /home/prow/go/pkg/mod/github.com/onsi/gomega@v1.42.0/internal/async_assertion.go:145
        > github.com/gardener/gardener/test/utils/rotation.(*GardenletKubeconfigRotationVerifier).After(0x28a8a5b24460, {0x35c2948?, 0x28a8a781ee10?}, 0x0)
            /home/prow/go/src/github.com/gardener/gardener/test/utils/rotation/gardenlet_kubeconfig.go:111
              | 
              | 	return fmt.Errorf("kubeconfig secret has not yet been renewed, timeBeforeRotation: %s, newClientCertificateIssuedAt: %s", v.timeBeforeRotation, newClientCertificateIssuedAt)
              > }).WithPolling(5 * time.Second).Should(Succeed())
```

seems like this test is constantly failing. not sure how this would be related to the VPN change. the tests don't even deploy HA VPN.

--- a/imagevector/containers.yaml
+++ b/imagevector/containers.yaml
@@ -167,7 +167,7 @@ images:
   - name: vpn-server
     sourceRepository: github.com/gardener/vpn2
     repository: europe-docker.pkg.dev/gardener-project/releases/gardener/vpn-server
-    tag: "0.49.0"
+    tag: "0.50.1"
   # OpenTelemetry
   - name: opentelemetry-operator
     sourceRepository: github.com/open-telemetry/opentelemetry-operator
@@ -457,7 +457,7 @@ images:
   - name: vpn-client
     sourceRepository: github.com/gardener/vpn2
     repository: europe-docker.pkg.dev/gardener-project/releases/gardener/vpn-client
-    tag: "0.49.0"
+    tag: "0.50.1"
   - name: coredns
     sourceRepository: github.com/coredns/coredns
     repository: registry.k8s.io/coredns/coredns

--- a/pkg/component/networking/vpn/shoot/shoot.go
+++ b/pkg/component/networking/vpn/shoot/shoot.go
@@ -654,11 +654,29 @@ func (v *vpnShoot) tunnelControllerContainer() *corev1.Container {
 		Name:            "tunnel-controller",
 		Image:           v.values.Image,
 		ImagePullPolicy: corev1.PullIfNotPresent,
-		Command:         []string{"/bin/tunnel-controller"},
+		Command:         []string{"/bin/vpn-client", "tunnel-controller"},
 		Resources: corev1.ResourceRequirements{
 			Requests: corev1.ResourceList{
 				corev1.ResourceCPU:    resource.MustParse("10m"),
 				corev1.ResourceMemory: resource.MustParse("10Mi"),
+			},
+		},
+		Env: []corev1.EnvVar{
+			{
+				Name:  "HA_VPN_CLIENTS",
+				Value: strconv.Itoa(v.values.HighAvailabilityNumberOfShootClients),
+			},
+			{
+				Name:  "WATCHDOG_WINDOW_SIZE",
+				Value: "20",
+			},
+			{
+				Name:  "WATCHDOG_THRESHOLD",
+				Value: "10",
+			},
+			{
+				Name:  "WATCHDOG_COOLDOWN",
+				Value: "20",
 			},
 		},
 		SecurityContext: &corev1.SecurityContext{

--- a/pkg/component/networking/vpn/shoot/shoot_test.go
+++ b/pkg/component/networking/vpn/shoot/shoot_test.go
@@ -694,7 +694,7 @@ var _ = Describe("VPNShoot", func() {
 					obj.Spec.Containers = append(obj.Spec.Containers, corev1.Container{
 						Name:    "tunnel-controller",
 						Image:   image,
-						Command: []string{"/bin/tunnel-controller"},
+						Command: []string{"/bin/vpn-client", "tunnel-controller"},
 						SecurityContext: &corev1.SecurityContext{
 							Privileged:               new(false),
 							AllowPrivilegeEscalation: new(false),
@@ -706,6 +706,24 @@ var _ = Describe("VPNShoot", func() {
 							Requests: corev1.ResourceList{
 								corev1.ResourceCPU:    resource.MustParse("10m"),
 								corev1.ResourceMemory: resource.MustParse("10Mi"),
+							},
+						},
+						Env: []corev1.EnvVar{
+							{
+								Name:  "HA_VPN_CLIENTS",
+								Value: "2",
+							},
+							{
+								Name:  "WATCHDOG_WINDOW_SIZE",
+								Value: "20",
+							},
+							{
+								Name:  "WATCHDOG_THRESHOLD",
+								Value: "10",
+							},
+							{
+								Name:  "WATCHDOG_COOLDOWN",
+								Value: "20",
 							},
 						},
 						ImagePullPolicy: corev1.PullIfNotPresent,

